contracts/DAITokenWrapper.sol

```diff
@@ -0,0 +1,150 @@
+     * @param r Part of the owner's signatue
+     * @param s Part of the owner's signatue
+     */
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) 
+        external
+        override
+        notZeroAddress(owner)
+        notZeroAddress(spender)
+        whenNotPaused
+    {
+        require(deadline == 0 || block.timestamp <= deadline, "PERMIT_EXPIRED");
+        require(r != bytes32(0) && s != bytes32(0), "INVALID_SIGNATURE_COMPONENTS");
+        uint nonce =  IDAI(daiTokenAddress).nonces(owner);
+        IDAI(daiTokenAddress).permit(owner, spender, nonce, deadline, true, v, r, s);
+        emit LogPermitCalledOnToken(daiTokenAddress, owner, spender, 0);    
+    }
+
+    /**
+     * @notice Set the address of the wrapper contract for the token. The wrapper is used to, for instance, allow the Boson Protocol functions that use permit functionality to work in a uniform way.
+     * @param _tokenAddress Address of the token which will be updated.
+     */
+    function setTokenAddress(address _tokenAddress)
+        external
+        override
+        onlyOwner
+        whenNotPaused
+        notZeroAddress(_tokenAddress)
+    {
+        daiTokenAddress = _tokenAddress;
+        emit LogTokenAddressChanged(_tokenAddress, owner());
+    }
+
+    /**
+     * @notice Get the address of the token wrapped by this contract
+     * @return Address of the token wrapper contract
+     */
+    function getTokenAddress()
+        external
+        view
+        override
+        returns (address)
+    {
+        return daiTokenAddress;
+    }
+
+    /**
+     * @notice Pause the Cashier && the Voucher Kernel contracts in case of emergency.
+     * All functions related to creating new batch, requestVoucher or withdraw will be paused, hence cannot be executed.
+     * There is special function for withdrawing funds if contract is paused.
+     */
+    function pause() 
+        external 
+        override
+        onlyOwner
+    {
+        
+        _pause();
+    }
+
+    /**
+     * @notice Unpause the Cashier && the Voucher Kernel contracts.
+     * All functions related to creating new batch, requestVoucher or withdraw will be unpaused.
+     */
+    function unpause() 
+        external 
+        override
+        onlyOwner
+    {
+        _unpause();
+    }
+
+    function nonces(address owner) external override view returns (uint256)
+    {
+        return IDAI(daiTokenAddress).nonces(owner);
+    }
+}
+
+/**
+ * @title IDAI
+ * @notice Interface for the purpose of calling the permit function on the deployed DAI token
+ */
+interface IDAI {
+ 
+    function name() external pure returns (string memory);
+    
+    function permit(address holder, address spender, uint256 nonce, uint256 expiry,
+                    bool allowed, uint8 v, bytes32 r, bytes32 s) external;
+
+    function transferFrom(
```

Comment:
I think we don't need to define the transferFrom() method because we don't call it

Reply:
It's needed for unit testing. If it's not defined in the interface, it can't be used for the mock. The mock needs to return a value for the transferFrom, or the unit tests won't work. It gets set like this: ` await mockDAI.mock.transferFrom`

Reply:
OK